### PR TITLE
Specialized mod exponentiation for base 2 in tfm.c and integer.c

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: wolfcrypt/src/integer.c [hunk 1]
```diff
@@ -2198,6 +2204,161 @@ LBL_RES:mp_clear (&res);
...
+#ifdef WOLFSSL_SMALL_STACK
+  res = (mp_int*)XMALLOC(sizeof(mp_int), NULL, DYNAMIC_TYPE_TMP_BUFFER);
+  if (res == NULL) {
+     return MP_MEM;
+  }
+#endif
+
+  /* now setup montgomery  */
+  if ((err = mp_montgomery_setup(P, &mp)) != MP_OKAY) {
+     return err;
```
Comment (toddouska) [inline wolfcrypt/src/integer.c]: What about res in SMALL_STACK case?

File: wolfcrypt/src/integer.c [hunk 2]
```diff
@@ -2198,6 +2204,161 @@ LBL_RES:mp_clear (&res);
...
+  }
+
+  /* setup result */
+  mp_init(res);
+
+  /* now we need R mod m */
+  mp_montgomery_calc_normalization(res, P);
```
Comment (toddouska) [inline wolfcrypt/src/integer.c]: What about return code failures?

File: wolfcrypt/src/integer.c [hunk 3]
```diff
@@ -2198,6 +2204,161 @@ LBL_RES:mp_clear (&res);
...
+        if (err != MP_OKAY) {
+        #ifdef WOLFSSL_SMALL_STACK
+          XFREE(res, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+        #endif
+          return err;
+        }
+      }
+
+      /* then multiply by 2^bitbuf */
+      mp_mul_2d(res, bitbuf, res);
```
Comment (toddouska) [inline wolfcrypt/src/integer.c]: return code check?

File: wolfcrypt/src/integer.c (hunk 1 above)
Comment (SparkiDev) [reply wolfcrypt/src/integer.c]: Fixed to jump to label.

File: wolfcrypt/src/integer.c (hunk 2 above)
Comment (SparkiDev) [reply wolfcrypt/src/integer.c]: Fixed to jump to label.

File: wolfcrypt/src/integer.c (hunk 3 above)
Comment (SparkiDev) [reply wolfcrypt/src/integer.c]: Fixed to do more checks and they jump to a label.

**Pull request merged**